packages/components/autocomplete/src/use-autocomplete.ts

```diff
@@ -450,6 +467,31 @@ export function useAutocomplete<T extends object>(originalProps: UseAutocomplete
           ),
         }),
       },
+      shouldCloseOnInteractOutside: (element: any) => {
+        let trigger = inputWrapperRef?.current;
+
+        // check if the current autocomplete is within the modal
+        const isWithinModal = element?.children?.[0]?.getAttribute("role") === "dialog" ?? false;
+
+        // if interacting outside the autocomplete
+        if (!trigger || !trigger.contains(element)) {
+          // blur the input in case it is currently focused
+          setShouldFocus(false);
+          // close the listbox close the listboxif it is not clicking overlay
+          // e.g. clicking another autocomplete
+          if (!isWithinModal) {
+            state.close();
+          }
+        } else {
+          // otherwise the autocomplete input should keep focused
+          setShouldFocus(true);
+        }
+
+        // if the autocomplete is in modal,
+        // clicking the overlay should close the listbox instead of closing the modal
+        // otherwise, allow interaction with other elements
+        return isWithinModal;
+      },
```

Comment:
@wingkwong could you please check whether other components that use the `Popover` under the hood have the same issues? 

- Dropdown
- DatePicker
- DateRangePicker
- Popover